### ISSUE TITLE
`pgpainless-cli verify` incorrectly accepts `INLINESIGNED` messages as `SIGNATURES` objects

### ISSUE THREAD

**Issue by dkg** [vanitasvitae/pgpainless#460]
`sop verify` is meant to verify detached signatures. That is, it compares a text or binary data against a `SIGNATURES` object, with reference to a set of acceptable signers.

`pgpainless-cli` is willing to treat an inline-signed message as a `SIGNATURES` object in this case. I don't think it should do that, because it is effectively confusing the type of object it receives with the type that it expects (see also [§11.4](https://www.ietf.org/archive/id/draft-dkg-openpgp-stateless-cli-11.html#optional-input-armoring).

Below is a transcript that demonstrates the concern. every command in it succeeds, but I think the final command (`pgpainless-cli verify`) should have failed.

Note that i'm using `sqop inline-sign` to make the inline-signed message here, because if i use `pgpainless-cli inline-sign`, it creates a compressed message, and a compressed `INLINESIGNED` message _doesn't_ get mis-read as a `SIGNATURES` object.

```
0 dkg@bob:~$ pgpainless-cli version --extended
PGPainless-SOP 1.6.8
https://codeberg.org/PGPainless/pgpainless/src/branch/master/pgpainless-sop

Implementation of the Stateless OpenPGP Protocol Version 07
https://datatracker.ietf.org/doc/html/draft-dkg-openpgp-stateless-cli-07

Based on pgpainless-core 1.6.8
https://pgpainless.org

Using Bouncy Castle 1.77
https://www.bouncycastle.org/java.html
0 dkg@bob:~$ sqop version --extended
sqop 0.36.0
sop-rs 0.8.0
Nettle 3.10 (Cv448: true, OCB: true)
Sequoia 1.21.2
0 dkg@bob:~$ pgpainless-cli generate-key test > test.key
0 dkg@bob:~$ pgpainless-cli extract-cert < test.key > test.cert
0 dkg@bob:~$ echo test | sqop inline-sign test.key > test.msg
0 dkg@bob:~$ pgpainless-cli inline-verify test.cert < test.msg
test
0 dkg@bob:~$ echo test | pgpainless-cli verify test.msg test.cert
2024-12-04T22:55:08Z 8FC0C899C8F11FCA6A70774734D6A228981EECBE 22726D5BA01E6DD570EB5FC5F542F5CD2B244E95 mode:binary
0 dkg@bob:~$ 
```

**Comment by vanitasvitae** [comment]
This is again due to PGPainless using the same API for all types of message verification and comes with the cost of having a simple API. Imho, users should not know the difference between inline- and detached signatures.

**Comment by dkg** [comment]
I agree with you, we want to keep things as simple for users as possible.  That said, I don't think it's possible for users to stay ignorant about what data type they're handling, unfortunately.

They need to know the difference between a certificate and a TSK (so that they don't send the TSK to someone else!) and they need to know the difference between an encrypted (and possibly signed) message and an message that happens to be signed (but is not encrypted).  And, they need to know the difference between a message that is signed, and a detached signature.

We already have actively dangerous APIs in the wild for OpenPGP signature verification, we don't need any more: `gpg` (and `gpgv`) attempted to make it so users wouldn't have to know the difference between a signed message and an unsigned message, and ended up here (from `gpgv(1)`):

> - `gpgv pgpfile`
> - `gpgv sigfile [datafile]`
> Verify the signature of the file. The second form is used for detached signatures, where `sigfile` is the detached signature (either ASCII-armored or binary) and `datafile` contains the signed data; if `datafile` is "`-`" the signed data is expected on stdin; if `datafile` is not given the name of the file holding the signed data is constructed by cutting off the extension ("`.asc`", "`.sig`" or "`.sign`") from `sigfile`.

Note that the results from invoking `gpgv` with a single argument will vary depending on whether *another* file exists that happens to have a similar name to the name of the single argument.

Consequently, if i know (or can control) the contents of a single file in the directory you're going to run gpgv from, i can hand you something that i claim is a signed message, and as long as `gpgv` is willing to interpret it as a `sigfile`, and you retain the filename i've given you, you'll think that you have a valid signed message, but actually the thing is a detached signature.  This is dangerous.

The interface should be as simple as possible, i agree: but _no simpler_!
